File: TP01/PalindromoRec.cpp
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>


bool isFim (char s[]);
bool EhPalind (char s[], int i, int j);
int main ()
{
	char entrada[2000];
   int eh;
   
   do{
         fgets(entrada, 200, stdin);
      
      if (!(isFim(entrada)))
      {
         eh = EhPalind (entrada, 0, strlen(entrada)-2);
         if (eh == 1)
         {
             printf ("SIM\n");
         }else{
            printf ("NAO\n");
         }
	   }

	}while (!(isFim(entrada)));
}
bool isFim (char s[])
{
	return (strlen(s) >= 3 && s[0] == 'F' && s[1] == 'I' && s[2] == 'M');

}
bool EhPalind (char s[], int i, int j)
{
   bool eh = true;
   
  
   if (i < j)
   {
      if(s[i] != s[j])
      {
         eh = false;
         i = j;
      }else{
            eh = EhPalind(s, ++i, --j);
      }  
  }
   
   return eh; 
}

File: TP01/PalindromoRec.java
class PalindromoRec
{
	public static boolean isFim (String s)
	{
		return (s.length() >= 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
	}


	public static boolean Palindromo (String s, int i, int j)
	{
	
		boolean resp = true;
		if (s.charAt(i) == s.charAt(j) && i < j)
		{
			return (Palindromo(s, ++i, --j));
		}else if (s.charAt(i) != s.charAt(j))
		{
			resp = false;
		}
		
		
		return (resp);
			
	}

	public static void main (String[] args)
	{
		String[] entrada = new String[1000];
		int qnt_palavras = 0;
		
		do{
			entrada[qnt_palavras] = MyIO.readLine();
		
		}while (!(isFim(entrada[qnt_palavras++])));
		qnt_palavras--;

		for (int i = 0; i < qnt_palavras; i++)
		{
			

			if (Palindromo(entrada[i], 0, entrada[i].length() - 1))
			{
				MyIO.println("SIM");
			}else{
				MyIO.println("NAO");
			}
		}
	
	
	
	}
}














